### PR TITLE
Fix smart task start action response dimmer attribute

### LINKED ISSUE #610: "1 validation error for SmartTaskResponse"
### Where are you using pytradfri (eg stand-alone, Home Assistant etc)

I've just installed pytrafri inside an Ubuntu Docker container, `coap-client` was installed following the notes in the README.

### Version of pytradfri

```
pytradfri==11.0.0
```

### Backend used (aiocoap, libcoap)

Initially I missed the notes in the README wrt building from source, so while I have this installed I'm not sure it is necessary or used.
```
ii  libcoap3:amd64              4.3.0-2build1                           amd64        C-Implementation of CoAP - libraries API version 3
```

The version of libcoap I built from source is
```
commit 25863042ae1e95138776f65c772f9fa6ce60495c (grafted, HEAD -> dtls, origin/dtls)
Author: Olaf Bergmann <bergmann@tzi.org>
Date:   Tue Apr 18 16:44:35 2017 +0200
```

### What type of environment are you in, eg Dockerized, Raspberry PI, VirtualBox, WiFi network (give as much details as possible)

```
docker run -it ubuntu:latest bash -l
```

followed by installing `python3` and `python3-pip`, followed by the dependencies.

### Expected behaviour

When I ran
```bash
python3 -i -m pytradfri 192.168.126.238
```
I expected to have the ability to modify the on/off schedule for an outlet.

### Actual behaviour

I get this traceback and no indication on how to configure the `Hot Water Recirculator` outlet.

```
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15001']
DEBUG:pytradfri.api.libcoap_api:Received: [65538,65551,65549,65548,65553,65552,65542,65541]
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15001', 65538]
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"Logan","9003":65538,"9002":1605168305,"9020":1648993072,"9054":1,"9019":1,"3":{"0":"IKEA of Sweden","1":"SYMFONISK Sound Controller","2":"","3":"2.1.024","6":3,"7":4554,"8":0,"9":5},"5750":8,"15018":[{"9003":0}],"15017":{"9115":["RINCON_347E5CF17AE201400"],"9123":0,"9117":0}}
<class 'pytradfri.device.DeviceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15001', 65551]
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"Living Room","9003":65551,"9002":1627611458,"9020":1673752021,"9054":0,"9019":1,"3":{"0":"IKEA of Sweden","1":"TRADFRI signal repeater","2":"","3":"2.3.086","6":1,"7":4354,"8":0},"5750":6,"15014":[{"9003":0}]}
<class 'pytradfri.device.DeviceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15001', 65549]
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"Dining Room","9003":65549,"9002":1613332595,"9020":1673752142,"9054":0,"9019":1,"3":{"0":"IKEA of Sweden","1":"TRADFRI signal repeater","2":"","3":"2.3.086","6":1,"7":4354,"8":0},"5750":6,"15014":[{"9003":0}]}
<class 'pytradfri.device.DeviceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15001', 65548]
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"Bedroom","9003":65548,"9002":1613332526,"9020":1673752154,"9054":0,"9019":1,"3":{"0":"IKEA of Sweden","1":"TRADFRI signal repeater","2":"","3":"2.3.086","6":1,"7":4354,"8":0},"5750":6,"15014":[{"9003":0}]}
<class 'pytradfri.device.DeviceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15001', 65553]
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"Hot Water Recirculator","9003":65553,"9002":1628093129,"9020":1673752176,"9054":0,"9019":1,"3":{"0":"IKEA of Sweden","1":"TRADFRI control outlet","2":"","3":"2.3.089","6":1,"7":4353,"8":0},"5750":3,"3312":[{"5850":1,"5849":0,"5851":254,"9003":0}]}
<class 'pytradfri.device.DeviceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15001', 65552]
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"Recirculator Switch","9003":65552,"9002":1628092700,"9020":1628109991,"9054":0,"9019":1,"3":{"0":"IKEA of Sweden","1":"TRADFRI on/off switch","2":"","3":"2.2.010","6":3,"7":4549,"8":0,"9":87},"5750":0,"15009":[{"9003":0}]}
<class 'pytradfri.device.DeviceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15001', 65542]
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"Black","9003":65542,"9002":1613328354,"9020":1616740694,"9054":0,"9019":1,"3":{"0":"IKEA of Sweden","1":"SYMFONISK Sound Controller","2":"","3":"2.1.024","6":3,"7":4554,"8":0,"9":60},"5750":8,"15018":[{"9003":0}],"15017":{"9115":["RINCON_48A6B8438D3301400"],"9123":0,"9117":0}}
<class 'pytradfri.device.DeviceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15001', 65541]
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"White","9003":65541,"9002":1613328308,"9020":1617815916,"9054":0,"9019":1,"3":{"0":"IKEA of Sweden","1":"SYMFONISK Sound Controller","2":"","3":"2.1.024","6":3,"7":4554,"8":0,"9":100},"5750":8,"15018":[{"9003":0}],"15017":{"9115":["RINCON_5CAAFDEFAAAA01400"],"9123":0,"9117":0}}
<class 'pytradfri.device.DeviceResponse'>
No lights found!
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15004']
DEBUG:pytradfri.api.libcoap_api:Received: [131076,131083,131078,131082]
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15004', 131076]
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"SuperGroup","9002":1605944776,"9003":131076,"5850":0,"5851":0,"9039":196617,"9108":2,"9018":{"15002":{"9003":[65553]}}}
<class 'pytradfri.group.GroupResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15004', 131083]
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"Garage","9002":1628092701,"9003":131083,"5850":0,"5851":0,"9039":196636,"9108":0,"9018":{"15002":{"9003":[65552,65553]}}}
<class 'pytradfri.group.GroupResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15004', 131078]
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"TRADFRI group 3","9002":1613328354,"9003":131078,"5850":0,"5851":0,"9039":196621,"9108":0,"9018":{"15002":{"9003":[65542]}}}
<class 'pytradfri.group.GroupResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15004', 131082]
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"TRADFRI group 4","9002":1613334116,"9003":131082,"5850":0,"5851":0,"9039":196633,"9108":0,"9018":{"15002":{"9003":[65541]}}}
<class 'pytradfri.group.GroupResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15005', '131076']
DEBUG:pytradfri.api.libcoap_api:Received: [196617]
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15005', '131076', 196617]: 131076
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"ALLOFF","9109":0,"9002":1605944776,"9003":196617,"9057":0,"9068":3,"15020":[{"15017":{"9115":[],"9123":3,"9117":10}}]}
<class 'pytradfri.resource.ApiResourceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15005', '131083']
DEBUG:pytradfri.api.libcoap_api:Received: [196636,196637,196638]
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15005', '131083', 196636]: 131083
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"EVERYDAY","9109":0,"9002":1628092702,"9003":196636,"9057":0,"9068":1,"15021":[{"5850":1,"9003":65553}]}
<class 'pytradfri.resource.ApiResourceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15005', '131083', 196637]: 131083
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"RELAX","9109":0,"9002":1628092702,"9003":196637,"9057":1,"9068":1,"15021":[{"5850":1,"9003":65553}]}
<class 'pytradfri.resource.ApiResourceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15005', '131083', 196638]: 131083
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"FOCUS","9109":0,"9002":1628092702,"9003":196638,"9057":2,"9068":1,"15021":[{"5850":1,"9003":65553}]}
<class 'pytradfri.resource.ApiResourceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15005', '131078']
DEBUG:pytradfri.api.libcoap_api:Received: [196621,196622,196623]
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15005', '131078', 196621]: 131078
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"EVERYDAY","9109":0,"9002":1613328355,"9003":196621,"9057":0,"9068":1}
<class 'pytradfri.resource.ApiResourceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15005', '131078', 196622]: 131078
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"RELAX","9109":0,"9002":1613328355,"9003":196622,"9057":1,"9068":1}
<class 'pytradfri.resource.ApiResourceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15005', '131078', 196623]: 131078
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"FOCUS","9109":0,"9002":1613328355,"9003":196623,"9057":2,"9068":1}
<class 'pytradfri.resource.ApiResourceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15005', '131082']
DEBUG:pytradfri.api.libcoap_api:Received: [196633,196634,196635]
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15005', '131082', 196633]: 131082
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"EVERYDAY","9109":0,"9002":1613334117,"9003":196633,"9057":0,"9068":1}
<class 'pytradfri.resource.ApiResourceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15005', '131082', 196634]: 131082
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"RELAX","9109":0,"9002":1613334117,"9003":196634,"9057":1,"9068":1}
<class 'pytradfri.resource.ApiResourceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15005', '131082', 196635]: 131082
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"FOCUS","9109":0,"9002":1613334117,"9003":196635,"9057":2,"9068":1}
<class 'pytradfri.resource.ApiResourceResponse'>
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15010']
DEBUG:pytradfri.api.libcoap_api:Received: [318615]
DEBUG:pytradfri.api.libcoap_api:Executing 192.168.126.238 get ['15010', 318615]
DEBUG:pytradfri.api.libcoap_api:Received: {"9001":"Light and dark","9002":1613335145,"9003":318615,"5850":1,"9040":2,"9041":127,"9042":{"5850":1,"15013":[{"9003":65553}]},"9043":{"5850":0,"15013":[{"9003":65553}]},"9044":[{"9046":15,"9047":0,"9048":7,"9049":0,"9226":0}]}
<class 'pytradfri.smart_task.SmartTaskResponse'>
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.10/dist-packages/pytradfri/__main__.py", line 97, in <module>
    tasks = api(tasks_commands)
  File "/usr/local/lib/python3.10/dist-packages/pytradfri/api/libcoap_api.py", line 131, in request
    result = self._execute(api_command, timeout=timeout)
  File "/usr/local/lib/python3.10/dist-packages/pytradfri/api/libcoap_api.py", line 106, in _execute
    api_command.process_result(_process_output(return_value, parse_json))
  File "/usr/local/lib/python3.10/dist-packages/pytradfri/command.py", line 62, in process_result
    self._result = self._process_result(result)
  File "/usr/local/lib/python3.10/dist-packages/pytradfri/gateway.py", line 251, in process_result
    return SmartTask(self, result)
  File "/usr/local/lib/python3.10/dist-packages/pytradfri/smart_task.py", line 112, in __init__
    super().__init__(raw)
  File "/usr/local/lib/python3.10/dist-packages/pytradfri/resource.py", line 40, in __init__
    self.raw = self._model_class(**raw)
  File "pydantic/main.py", line 342, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for SmartTaskResponse
9042 -> 15013 -> 0 -> 5851
  field required (type=value_error.missing)
```

I don't know to what degree the traceback I got is preventing me from proceeding, but I figured as I'm poking around I can try to report back.

### PR DESCRIPTION
**Pull request by blast-hardcheese**
Resolves #610

Also fleshed out some examples for how to interact with `SmartTask`s, happy to revert some or all of those if undesired.

Thanks for this library!

### CODE REVIEW

File: pytradfri/__main__.py [hunk 1]
```diff
@@ -122,12 +130,17 @@ def dump_devices() -> None:
...
         print("> light.light_control.lights")
         print("> api(light.light_control.set_dimmer(10))")
         print("> api(light.light_control.set_dimmer(254))")
         print("> api(light.light_control.set_xy_color(30015, 26870))")
         print("> api(light.light_control.set_predefined_color('Warm Amber'))")
         print("> api(lights[1].light_control.set_dimmer(20))")
+    if socket:
+        print("> sockets")
+        print("> socket.socket_control.sockets")
+        print("> api(socket.socket_control.set_values({ const.ATTR_REPEAT_DAYS: 0b1111111 }))")
```
Comment (MartinHjelmare) [inline pytradfri/__main__.py:143]: We don't use the const module above.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -6,7 +6,8 @@ name: Run Tests
 on:
   push:
     branches: [master]
-  pull_request:
+  pull_request: {}
+  workflow_dispatch: {}
```
Comment (MartinHjelmare) [inline .github/workflows/test.yml:10]: Please don't change this. It's unrelated.

File: pytradfri/__main__.py (hunk 1 above)
Comment (ggravlingen) [reply pytradfri/__main__.py]: Do we need to change `__main__.py` to fix this bug? I don't mind appending this file, as it makes the library easier to use, but perhaps in another PR?

**Comment by ggravlingen** [comment]
The linting errors in the 3.9 run seem unrelated to the changes made here. Should I resolve this in another PR @MartinHjelmare?

```
  lint: commands[2]> pylint examples pytradfri tests
  ************* Module example_async
  examples/example_async.py:26:0: I0021: Useless suppression of 'import-error' (useless-suppression)
  ************* Module example_cover_async
  examples/example_cover_async.py:26:0: I0021: Useless suppression of 'import-error' (useless-suppression)
  ************* Module debug_info
  examples/debug_info.py:28:0: I0021: Useless suppression of 'import-error' (useless-suppression)
  ************* Module example_pair
  examples/example_pair.py:31:0: I0021: Useless suppression of 'import-error' (useless-suppression)
  examples/example_pair.py:137:0: I0021: Useless suppression of 'unsupported-membership-test' (useless-suppression)
  ************* Module example_sync
  examples/example_sync.py:27:0: I0021: Useless suppression of 'import-error' (useless-suppression)
  ************* Module example_socket_async
  examples/example_socket_async.py:26:0: I0021: Useless suppression of 'import-error' (useless-suppression)
  ************* Module example_air_purifier_async
  examples/example_air_purifier_async.py:26:0: I0021: Useless suppression of 'import-error' (useless-suppression)
  ************* Module pytradfri.__main__
  pytradfri/__main__.py:9:0: W0611: Unused const imported from pytradfri (unused-import)
  pytradfri/__main__.py:9:0: I0021: Useless suppression of 'import-error' (useless-suppression)

```

File: .github/workflows/test.yml (hunk 2 above)
Comment (blast-hardcheese) [reply .github/workflows/test.yml]: I pushed this incorrectly -- I have reverted it.

File: pytradfri/__main__.py (hunk 1 above)
Comment (blast-hardcheese) [reply pytradfri/__main__.py]: I can do it in another PR, yes. I have been iterating just trying to get the tests to pass, I didn't realize there was already activity in this PR. I've stopped updating so far, but happy to move it out.

**Comment by blast-hardcheese** [comment]
So, the underlying bug was resolved with [773aca7](https://github.com/home-assistant-libs/pytradfri/pull/614/commits/773aca7a5280752732ab476b00096a8306e9dc1a).

I think https://github.com/home-assistant-libs/pytradfri/pull/614/commits/6b7e12cf4454414819fabf68c43c3eb9797df87b fits in line with this library, and is the underlying task that I was trying to achieve with my project here, so I figured adding it made sense.

File: pytradfri/__main__.py (hunk 1 above)
Comment (blast-hardcheese) [reply pytradfri/__main__.py]: OK. I split out #615 from this. I will force push to reduce down to just the bugfix.

**Comment by blast-hardcheese** [comment]
OK, this and #615 are ready for review. Sorry for the noise, I was having difficulty getting the tests running locally, I was using GitHub Actions to run tests in my fork.

File: tests/test_smart_task.py [hunk 3]
```diff
@@ -55,6 +55,18 @@
...
 
+TASK_OPTIONAL_DIMMER = {
+    "9001": "Light and dark",
+    "9002": 1613335145,
+    "9003": 318615,
+    "5850": 1,
+    "9040": 2,
+    "9041": 127,
+    "9042": {"5850": 1, "15013": [{"9003": 65553}]},
+    "9043": {"5850": 0, "15013": [{"9003": 65553}]},
```
Comment (MartinHjelmare) [inline tests/test_smart_task.py:66]: This is the response captured in the issue. Do we know what 9043 represents? We don't have that one in our `const` module.

Comment (blast-hardcheese) [reply tests/test_smart_task.py:66]: I'm unable to test using the app for the next few weeks unfortunately, I suspect it relates to controlling an outlet via smart task. Is there a way I can confirm this somehow, given my setup?

**Review by MartinHjelmare — APPROVED** [review]
Thanks!

**Pull request merged**